mm2src/mm2_bitcoin/spv_validation/src/helpers_validation.rs

```diff
@@ -306,13 +313,83 @@ pub(crate) fn merkle_prove(
 
 fn validate_header_prev_hash(actual: &H256, to_compare_with: &H256) -> bool { actual == to_compare_with }
 
+fn validate_btc_max_stored_headers_value(max_stored_block_headers: u64) -> Result<(), SPVError> {
+    if RETARGETING_INTERVAL > max_stored_block_headers as u32 {
+        return Err(SPVError::StartingBlockHeaderError(format!(
+            "max_stored_block_headers {max_stored_block_headers} must be greater than retargeting interval {RETARGETING_INTERVAL}",
+        )));
+    }
+
+    Ok(())
+}
+
+fn validate_btc_starting_header_height(coin: &str, height: u64) -> Result<(), SPVError> {
+    let is_retarget = height % RETARGETING_INTERVAL as u64;
+    if is_retarget != 0 {
+        return Err(SPVError::WrongRetargetHeight {
+            coin: coin.to_string(),
+            expected_height: height - is_retarget,
+        });
+    }
+
+    Ok(())
+}
+
 /// SPV headers verification parameters
 #[derive(Clone, Debug, Deserialize)]
-pub struct BlockHeaderVerificationParams {
+pub struct BlockHeaderValidationParams {
     pub difficulty_check: bool,
     pub constant_difficulty: bool,
     pub difficulty_algorithm: Option<DifficultyAlgorithm>,
-    pub genesis_block_header: String,
+}
+
+#[derive(Debug, Clone, Deserialize)]
+#[serde(tag = "type")]
+pub enum StartingBlockHeader {
+    #[serde(rename = "genesis")]
+    Genesis { hex: String },
+    #[serde(rename = "other")]
+    Other { height: u64, hex: String },
+}
```

Comment:
As you @borngraced already discussed with me on DM, it's not very easy to get the block header hex for some coins. What we need from the header for now is the `hash`, `time` and `bits`, which are easy to get from any coin block explorer. We can add those fields here instead of the hex, and make the required changes.